src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1069,6 +1069,17 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return Promise.resolve(true);
 	}
 
+	/**
+	 * Replaces connection info uri with new uri.
+	 */
+	public changeConnectionUri(newUri: string, oldUri: string) {
+		this._connectionStatusManager.changeConnectionUri(newUri, oldUri);
+		if (this._uriToProvider[oldUri]) {
```

Comment:
Any reason you're not treating this like the others where it being missing from the map would be considered an error?